cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/AbstractConnectOperator.java

```diff
@@ -563,6 +566,91 @@ private boolean needsReconfiguring(Reconciliation reconciliation, String connect
+                            // Ignore restart failures - just try again on the next reconcile
+                            log.warn("{}: Failed to restart connector {}. {}", reconciliation, connectorName, throwable.getMessage());
+                            return Future.succeededFuture();
+                        })
+                    .compose(ignored -> apiClient.statusWithBackOff(new BackOff(200L, 2, 10), host, port,
+                        connectorName));
+        } else {
+            return Future.succeededFuture(status);
+        }
+    }
+
+    private Future<Map<String, Object>> maybeRestartConnectorTask(Reconciliation reconciliation, String host, KafkaConnectApi apiClient, String connectorName, CustomResource resource, Map<String, Object> status) {
+        int taskID = getRestartTaskAnnotationTaskID(resource, connectorName);
+        if (taskID >= 0) {
+            log.debug("{}: Restarting connector task {}:{}", reconciliation, connectorName, taskID);
+            return apiClient.restartTask(host, port, connectorName, taskID)
+                    .compose(ignored -> removeRestartTaskAnnotation(reconciliation, resource),
+                        throwable -> {
+                            // Ignore restart failures - just try again on the next reconcile
+                            log.warn("{}: Failed to restart connector task {}:{}. {}", reconciliation, connectorName, taskID, throwable.getMessage());
+                            return Future.succeededFuture();
+                        })
+                    .compose(ignored -> apiClient.statusWithBackOff(new BackOff(200L, 2, 10), host, port,
+                        connectorName));
+        } else {
+            return Future.succeededFuture(status);
+        }
+    }
+
+    /**
+     * Whether the provided resource instance is a KafkaConnector and has the strimzi.io/restart annotation
+     *
+     * @param resource resource instance to check
+     * @param resource connectorName name of the connector to check
+     * @return true if the provided resource instance has the strimzi.io/restart annotation; false otherwise
+     */
+    protected boolean hasRestartAnnotation(CustomResource resource, String connectorName) {
+        return Annotations.booleanAnnotation(resource, ANNO_STRIMZI_IO_RESTART, false);
+    }
+
+    /**
+     * Return the ID of the connector task to be restarted if the provided KafkaConnector resource instance has the strimzio.io/restart-task annotation
+     *
+     * @param resource resource instance to check
+     * @param connectorName KafkaConnector resource instance to check
+     * @return the ID of the task to be restarted if the provided KafkaConnector resource instance has the strimzio.io/restart-task annotation or -1 otherwise.
+     */
+    protected int getRestartTaskAnnotationTaskID(CustomResource resource, String connectorName) {
+        return Annotations.intAnnotation(resource, ANNO_STRIMZI_IO_RESTART_TASK, -1);
```

Comment:
I wonder if we should support a list of ids. It would make it simpler to restart several failed tasks.

Reply:
Thanks for the review. I have started looking into this - it should be possible to restart tasks on a `KafkaConnector` resource using an annotation like: `strimzi.io/restart-tasks: "0 1 2"`, although it does raise some questions:
 - what should happen if one of the `restart` calls fails (e.g. maybe there is no task `2`)? Should the annotation be altered to only include the task IDs that were not restarted, so that they can be attempted on the next reconcile?
 - what should the annotation look like for `KafkaMirrorMaker2` resources? The code currently supports annotations that address a single MM2 connector task like: `strimzi.io/restart-connector-tasks: "<MM2-connector-name>:<task-id>"`. Do you think it should allow multiple MM2 connectors to be specified? Or just multiple tasks for a single connector? For example `strimzi.io/restart-connector-tasks: "<MM2-connector-name>:0 1 2"`

Reply:
These are good points. I guess it might be a good idea to leave this idea for now and see how this stuff is used in practice. It seems like it would be easy enough to add at a later date.